designs/CB-OPT-design.md

```diff
@@ -0,0 +1,172 @@
+# Design of Optimization on Compiler B
+
+## A. Test cases / Scenarios
+> Analyses that extract useful guidance for potential transformations without altering the IR
+
+### 1. Liveness Analysis
```

Comment:
I'm not sure I would describe what you're doing here as liveness analysis. What variables are live? I expect the result of a live analysis to be a set annotation representing live variables.